concept/impl/AttributeTypeImpl.java

```diff
@@ -119,27 +119,32 @@ private void checkInstancesMatchRegex(@Nullable String regex) {
     }
 
     @Override
-    public Attribute<D> create(D value) {
+    public Attribute<D> create(Object value) {
         return putAttribute(value, false);
     }
 
     @Override
-    public Attribute<D> putAttributeInferred(D value) {
+    public Attribute<D> putAttributeInferred(Object value) {
         return putAttribute(value, true);
     }
 
     /**
      * Method through which all new instance creations of attributes pass through
      */
-    private Attribute<D> putAttribute(D value, boolean isInferred) {
+    private Attribute<D> putAttribute(Object value, boolean isInferred) {
         Objects.requireNonNull(value);
 
+        // check that the value conforms
         if (valueType().equals(ValueType.STRING)) checkConformsToRegexes((String) value);
+        D writeableValue = AttributeValueConverter.tryConvertForWrite(this, value);
 
-        Attribute<D> instance = getAttribute(value);
+        // if the value is readable, return read value
+        D readableValue = AttributeValueConverter.tryConvertForRead(this, value);
```

Comment:
What happens if the value is not readable?

Reply:
get us an exception!